### PR TITLE
chore: Add migration to update indexes for DB collection to include branchName param 

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> After git feature is introduced resources are uniquely identified by combining defaultResourceId and branchName. This PR creates the compound indexes for newPage, newAction, actionCollection and application collections to include branchName which will optimise the find query.   

## Type of change

- Enhancement

## How Has This Been Tested?

- Manual 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 1]
```diff
@@ -4873,4 +4873,115 @@ private Query getQueryToFetchAllPluginActionsWhichAreNotDeleted(Plugin plugin) {
...
+     * This migration introduces indexes on newAction, actionCollection, newPage and application collection to take
+     * branchName param into consideration for optimising the find query for git connected applications
+     */
+    @ChangeSet(order = "110", id = "update-index-for-git", author = "")
+    public void updateGitIndexes(MongockTemplate mongockTemplate) {
+
+        // Avoid duplicate key exception
+
+        // Update deleted pages for defaultPageId
+        final Query pageQuery = query(where(fieldName(QNewPage.newPage.deleted)).is(true));
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:4887]: Earlier only active documents were migrated, which was throwing duplicate exception as `defaultResourceId(null), branchName(null) and deleted(true)` fields were constant

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**